tests/test_imports.py

```diff
@@ -81,3 +82,19 @@ def test_cli_import(self):
         paths_above_threshold = get_paths_above_threshold(sorted_data, 0.05, max_depth=7)
         err_msg += f"\n{convert_list_to_string(paths_above_threshold)}"
         self.assertLess(pct_more, 20, err_msg)
+
+
+@require_transformer_engine
+class LazyImportTester(TempDirTestCase):
+    """
+    Test suite which checks if specific packages are lazy-loaded.
+
+    Eager-import will trigger circular import in some case,
+    e.g. in huggingface/accelerate#3056.
+    """
+
+    # @require_transformer_engine
+    def test_te_import(self):
+        output = run_import_time("import accelerate, accelerate.utils.transformer_engine")
+
+        self.assertFalse(' transformer_engine' in output, '`transformer_engine` should not be imported on import')
```

Comment:
We have no CI for transformerengine, so not sure this test is actually valuable imo

Reply:
Thanks for your review! I added this test sort of as a self-contained doc. But it's indeed not much valuable if not in CI. I've just add a one line comment in the other file, and I can undo the test if you think that's better.